projects/js-packages/charts/src/components/legend/hooks/use-chart-legend-items.ts

```diff
@@ -58,42 +63,47 @@ function formatPointValue(
 
 /**
  * Processes SeriesData into legend items
- * @param seriesData  - The series data to process
- * @param theme       - The chart theme for colors
- * @param showValues  - Whether to show values in legend
- * @param withGlyph   - Whether to include glyph rendering
- * @param glyphSize   - Size of the glyph
- * @param renderGlyph - Component to render the glyph
- * @param legendShape - The shape to use for the legend
+ * @param seriesData       - The series data to process
+ * @param getElementStyles - Function to get element styles
+ * @param showValues       - Whether to show values in legend
+ * @param withGlyph        - Whether to include glyph rendering
+ * @param glyphSize        - Size of the glyph
+ * @param renderGlyph      - Component to render the glyph
+ * @param legendShape      - The shape to use for the item
  * @return Array of processed legend items
  */
 function processSeriesData(
 	seriesData: SeriesData[],
-	theme: ChartTheme,
+	getElementStyles: ( params: GetElementStylesParams ) => ElementStyles,
 	showValues: boolean,
 	withGlyph: boolean,
 	glyphSize: number,
 	renderGlyph?: < Datum extends object >( props: GlyphProps< Datum > ) => ReactNode,
 	legendShape?: LegendShape< SeriesData[], number >
 ): BaseLegendItem[] {
 	const mapper = ( series: SeriesData, index: number ) => {
-		const { shapeStyles } = getItemShapeStyles( series, index, theme, legendShape );
-		const baseItem = {
+		const { color, glyph, shapeStyles } = getElementStyles( {
+			data: series,
+			index,
+			legendShape,
+		} );
+
+		const baseItem: BaseLegendItem = {
 			label: series.label,
 			value: showValues ? series.data?.length?.toString() || '0' : '',
-			color: getSeriesStroke( series, index, theme.colors ),
+			color,
 			shapeStyle: shapeStyles,
-			group: series.group,
-			index,
-			overrideColor: series.options?.stroke,
 		};
 
-		if ( withGlyph && renderGlyph ) {
-			return {
-				...baseItem,
-				glyphSize,
-				renderGlyph,
-			};
+		if ( withGlyph ) {
+			const glyphToUse = renderGlyph || glyph;
```

Comment:
This isn't working yet

Reply:
Fixed in [81a34b8](https://github.com/Automattic/jetpack/pull/45207/commits/81a34b8be0004913f958670592c7675484a6c1fc)